### DISCUSSION TITLE
Microtonal Tuxguitar / help setting up build file?

### DISCUSSION THREAD

**Discussion by afrolover666** [helge17/tuxguitar#186] [General]
Hi, I'm very new to all this and have little to no idea what I'm doing with Java (exactly what you wanted to hear, I know). But I want to know if it's possible to make microtonal notes in TuxGuitar. My simple idea is to view the fretboard in TuxGuitar, and make edits to the fretboard that divide the frets into microtonal scales (see King Gizzard and the Lizard Wizard - Microtonal Flying Banana). They built their own guitars with more frets in between the standard frets, and I want to replicate this on TuxGuitar. I figured that if I can divide the frets and assign customized note names, I'll be able to use them as long as the MIDI synthesizer allows for microtones. Or bypass the fretboard view totally and just write tabs like 1.5, 1.75, etc. instead of whole numbers for tabs. I am having some trouble getting started, not sure if it's even possible... Currently I can't even get the build-script to work on my Windows 10 machine and I'm not sure what I'm doing wrong. I figure I need the build script to work first before I delve into the microtonal editing (that way I can test my edits). I've ran the tuxguitar.bat file a few times, with changes in between, but cannot get it to work. 

![ApplicationFrameHost_CiIqheu9D6](https://github.com/helge17/tuxguitar/assets/155195769/e96df9ee-1103-4c7f-a0ab-7de45fb2345a)

After creating the missing jre/bin/ directory with my java executable inside of it, the error message changes to this:

![ApplicationFrameHost_1JutK0DKwc](https://github.com/helge17/tuxguitar/assets/155195769/ccd06ed4-80c5-4edd-8cb7-72e7bd8ab214)

I would be very grateful if anyone could point out what I'm doing wrong. Or if someone else has already made a microtonal edit for tuxguitar, but I have not been able to find one.

Thank you

**Comment by guiv42** [comment]
Hi
honestly, I do not really understand how you are trying to modify TuxGuitar. Such an evolution does not seem trivial to me.

> Currently I can't even get the build-script to work on my Windows 10 machine and I'm not sure what I'm doing wrong

Which build-script are you referring to? The process to build TuxGuitar is described in [install.md](https://github.com/helge17/tuxguitar/blob/master/INSTALL.md) file. Please note that this process is only documented *on Linux* , even to build the windows version. Currently I have no idea about how to build TuxGuitar directly from Windows.

**Reply by afrolover666** [reply]
Hi guiv42,

Thanks for such a quick reply. To clarify my intent, I want to have access to notes outside of the Western scale (think Turkish or Indian music with sitars). Quarter sharps, quarter flats, three-quarter sharps and three-quarter flats. I figure these notes are accessible, because you can bend to microtones using TuxGuitar's bend functionality. Mus2 is a program I found that has these built in, and it is similar to TuxGuitar.

I was attempting to build on Windows using the "\tuxguitar-master\desktop\build-scripts\common-resources\common-windows\tuxguitar.bat" file. I also noticed that there was no build instructions for Windows in the install.md file, so maybe I'm just trying to do something that's completely impossible. I don't know. 

If TuxGuitar cannot be built from the source code on Windows, is there any alternative method I could use to edit the existing installation's files? Before I found this Github repo, I was actually using JD-GUI to view my installed version of TuxGuitar's .CLASS files and recompiling them in Eclipse IDE. Again, not sure if that's possible. I suppose the alternative is to install a Linux VM and try building the Windows version from there to test on my native OS?

**Reply by guiv42** [reply]
> I was attempting to build on Windows using the "\tuxguitar-master\desktop\build-scripts\common-resources\common-windows\tuxguitar.bat" file.

This file is absolutely not related to the build process. It's a start script to launch the app on windows (included in the output of the build process).

> I also noticed that there was no build instructions for Windows in the install.md file, so maybe I'm just trying to do something that's completely impossible. I don't know.

I don't know either, but I have never heard of any build process on Windows. So, if you follow this path you will be on your own (i.e. we cannot provide support).

> is there any alternative method I could use to edit the existing installation's files? Before I found this Github repo, I was actually using JD-GUI to view my installed version of TuxGuitar's .CLASS files and recompiling them in Eclipse IDE. Again, not sure if that's possible

No idea!

> I suppose the alternative is to install a Linux VM and try building the Windows version from there to test on my native OS?

I would recommend this solution. First it looks much simpler to me, then if you face some issues in this configuration at least you should be able to find some help here.

**Comment by afrolover666** [comment]
> > I was attempting to build on Windows using the "\tuxguitar-master\desktop\build-scripts\common-resources\common-windows\tuxguitar.bat" file.
> 
> This file is absolutely not related to the build process. It's a start script to launch the app on windows (included in the output of the build process).
>

That explains a lot, lol. I was taking a shot in the dark thinking that it may have had everything included to build the Windows executable from all the other files in the repo and then run it. (Leading me to think I could change some .java files around, run the .bat, and launch the new version). My lack of coding experience is showing. It makes more sense to build on Linux following the install.md file, so I'll try that out first (and I'll probably come back with more noob questions). Thanks again.

**Comment by BeFrankly** [comment]
Dear @afrolover666 , I know what you mean and what you want. Turkish and Indian music is different from let me say European music, because it knows these quarter tonal steps. In the European orchestra where I play guitar, we had a project with an Oud Musicians from Syria. He told us a lot about his instrument. But as you also was thinking about is, that the MIDI software or let me say the MIDI standard must have these microtonal steps as well. I guess this industrial standard doesn't have this quarter steps. European Musicsoftware often has this limitations, I guess, because the music is limited. Arabic music has a lot more possibilities. 

Tuxguitar at the moment does not reflect arabic music. That's a shame but your post is a good impulse. If it is possible to find more developers it might be possible to implement this feature, but at the moment I do not see a way.

By the way: The name "Tuxguitar" contains the prefix "Tux" that belongs to the maskot of Linux, not Windows.

https://en.wikipedia.org/wiki/Tux_(mascot) 

@guiv42 , should we add a feature request?

**Reply by guiv42** [reply]
> The name "Tuxguitar" contains the prefix "Tux" that belongs to the maskot of Linux, not Windows

Yes, but Tux also plays on Windows ;)

> should we add a feature request?

I don't know if we *should*, but for sure we *can*
I haven't looked in details, but this seems a *very challenging* evolution. It does not seem reachable to me in a near future.

**Reply by BeFrankly** [reply]
I agree that we will not reach this in the near future. But if we do not create a feature request, this aspect will get lost. Nobody will remember this post in a few years.
